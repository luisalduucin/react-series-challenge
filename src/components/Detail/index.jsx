import React from "react";
import PropTypes from 'prop-types';
import {DetailWrapper} from "./Detail.style";
import GifCardContainer from "../../containers/GifCard";
import NotFound from "../NotFound";

const Detail = ({favorites, match, trending}) => {
  const obtainGifById = (id) => {
    let gif = favorites.find((favorite) => favorite.id === id);
    if (gif) {
      return {data: gif, isFavorite: true}
    }
    gif = trending.find((gif) => gif.id === id);
    return gif ? {data: gif, isFavorite: false} : null;
  };

  const gifCard = (gif) => {
    return (
      <GifCardContainer
        gif={gif.data}
        isFavorite={gif.isFavorite}
        isDetailBtnVisible={false}
        imageSize={'original'}
      />
    )
  };

  const gif = obtainGifById(match.params.id);

  return (
    <DetailWrapper>
      {
        gif ? gifCard(gif) : <NotFound />
      }
    </DetailWrapper>
  )
};

Detail.propTypes = {
  trending: PropTypes.array.isRequired,
  favorites: PropTypes.array.isRequired
};

export default Detail;
