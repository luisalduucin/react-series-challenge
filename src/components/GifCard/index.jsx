import React from 'react';
import PropTypes from 'prop-types';
import {Btn, BtnsWrapper, GifCardWrapper} from "./GifCard.style";
import eyeIcon from './assets/eye.png';
import heartBorder from './assets/heart-border.svg';
import heartFilled from './assets/heart-filled.svg';

const GifCard = ({gif, isFavorite, imageSize, toggleFavorite}) => {
  const image = imageSize === 'small' ? gif.images.small : gif.images.original;
  const heartIcon = isFavorite ? heartFilled : heartBorder;
  return (
    <GifCardWrapper>
      <img alt={'gif'} src={image} />
      <BtnsWrapper>
        <Btn alt={'open'} src={eyeIcon} />
        <Btn alt={'favorite'} src={heartIcon} onClick={toggleFavorite.bind(this, gif)} />
      </BtnsWrapper>
    </GifCardWrapper>
  )
};

GifCard.propTypes = {
  gif: PropTypes.shape({
    id: PropTypes.string,
    images: PropTypes.shape({
      original: PropTypes.string,
      small: PropTypes.string
    })
  }).isRequired,
  isFavorite: PropTypes.bool.isRequired,
  imageSize: PropTypes.oneOf(['small', 'original']).isRequired,
  toggleFavorite: PropTypes.func.isRequired
};

export default GifCard
