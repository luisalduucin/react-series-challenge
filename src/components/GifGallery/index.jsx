import React from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';
import {FavoriteBtn, HomeWrapper, SearchBtn, SearchInput, SearchWrapper, TrendingGifCards} from './GifGallery.style';
import searchIcon from './search.png';
import favoriteIcon from './heart-filled-white.png'
import GifCard from "../GifCard";

class GifGallery extends React.Component {
  componentWillMount() {
    if (this.props.dataSource === 'GIPHY') {
      this.props.fetchTrendingGifs();
    }
  }

  render() {
    const dataSource = this.props.dataSource === 'GIPHY' ? this.props.trending: this.props.favorites;
    return (
      <HomeWrapper>
        <SearchWrapper>
          <SearchInput placeholder={'Type to search cool GIFs'} />
          <SearchBtn src={searchIcon} />
          <FavoriteBtn src={favoriteIcon} onClick={() => this.props.history.push('/favorites')} />
        </SearchWrapper>
        <TrendingGifCards>
          {
            dataSource.map((gif) => {
              const isFavorite = !!this.props.favorites.find((favorite) => favorite.id === gif.id);
              return (
                <GifCard
                  gif={gif}
                  isFavorite={isFavorite}
                  imageSize={'small'}
                  key={gif.id}
                  toggleFavorite={this.props.toggleFavorite}
                 />
              )
            })
          }
        </TrendingGifCards>
      </HomeWrapper>
    );
  }
}

GifGallery.propTypes = {
  fetchTrendingGifs: PropTypes.func,
  toggleFavorite: PropTypes.func.isRequired,
  trending: PropTypes.array.isRequired
};

export default withRouter(GifGallery);
