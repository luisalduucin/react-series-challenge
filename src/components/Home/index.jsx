import React from 'react';
import PropTypes from 'prop-types';
import {HomeWrapper, SearchButton, SearchInput, SearchWrapper, TrendingGifCards} from './Home.style';
import searchIcon from './search.png';
import GifCard from "../GifCard";

class Home extends React.Component {
  componentWillMount() {
    this.props.fetchTrendingGifs();
  }

  render() {
    return (
      <HomeWrapper>
        <SearchWrapper>
          <SearchInput placeholder={'Type to search cool GIFs'} />
          <SearchButton src={searchIcon} />
        </SearchWrapper>
        <TrendingGifCards>
          {
            this.props.trending.map((gif) => {
              return (
                <GifCard url={gif.images.fixed_width_downsampled.url}  isFavorite={false} />
              )
            })
          }
        </TrendingGifCards>
      </HomeWrapper>
    );
  }
}

Home.propTypes = {
  fetchTrendingGifs: PropTypes.func.isRequired,
  trending: PropTypes.array.isRequired
};

export default Home;
