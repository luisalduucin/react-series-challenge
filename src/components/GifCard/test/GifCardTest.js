import React from 'react';
import renderer from 'react-test-renderer';
import {GifCard} from "../../../components/GifCard";

const mockGif = {
  id: 'p3n7NWvfz3u0c1meay',
  images: {
    original: 'https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/giphy.gif',
    small: 'https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/200w_d.gif'
  }
};

describe('GifCard component', () => {
  it('should render', () => {
    const component = renderer.create(
      <GifCard
        gif={mockGif}
        isFavorite={false}
        isDetailBtnVisible={false}
        imageSize={'small'}
        toggleFavorite={() => {}}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should not render visible btn', () => {
    const component = renderer.create(
      <GifCard
        gif={mockGif}
        isFavorite={false}
        isDetailBtnVisible
        imageSize={'small'}
        toggleFavorite={() => {}}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should render filled heart', () => {
    const component = renderer.create(
      <GifCard
        gif={mockGif}
        isFavorite
        isDetailBtnVisible
        imageSize={'small'}
        toggleFavorite={() => {}}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should render the big image', () => {
    const component = renderer.create(
      <GifCard
        gif={mockGif}
        isFavorite
        isDetailBtnVisible
        imageSize={'original'}
        toggleFavorite={() => {}}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
