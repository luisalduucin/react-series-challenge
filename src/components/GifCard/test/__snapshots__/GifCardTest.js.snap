// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifCard component should not render visible btn 1`] = `
<div
  className="sc-bwzfXH dOTfHK"
>
  <img
    alt="gif"
    src="https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/200w_d.gif"
  />
  <div
    className="sc-bdVaJa kWteoq"
  >
    <img
      alt="detail"
      className="sc-htpNat jSMwyr"
      onClick={[Function]}
      src={
        Object {
          "process": [Function],
        }
      }
    />
    <img
      alt="favorite"
      className="sc-htpNat jSMwyr"
      onClick={[Function]}
      src={
        Object {
          "process": [Function],
        }
      }
    />
  </div>
</div>
`;

exports[`GifCard component should render 1`] = `
<div
  className="sc-bwzfXH dOTfHK"
>
  <img
    alt="gif"
    src="https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/200w_d.gif"
  />
  <div
    className="sc-bdVaJa kWteoq"
  >
    <img
      alt="favorite"
      className="sc-htpNat jSMwyr"
      onClick={[Function]}
      src={
        Object {
          "process": [Function],
        }
      }
    />
  </div>
</div>
`;

exports[`GifCard component should render filled heart 1`] = `
<div
  className="sc-bwzfXH dOTfHK"
>
  <img
    alt="gif"
    src="https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/200w_d.gif"
  />
  <div
    className="sc-bdVaJa kWteoq"
  >
    <img
      alt="detail"
      className="sc-htpNat jSMwyr"
      onClick={[Function]}
      src={
        Object {
          "process": [Function],
        }
      }
    />
    <img
      alt="favorite"
      className="sc-htpNat jSMwyr"
      onClick={[Function]}
      src={
        Object {
          "process": [Function],
        }
      }
    />
  </div>
</div>
`;

exports[`GifCard component should render the big image 1`] = `
<div
  className="sc-bwzfXH dOTfHK"
>
  <img
    alt="gif"
    src="https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/giphy.gif"
  />
  <div
    className="sc-bdVaJa kWteoq"
  >
    <img
      alt="detail"
      className="sc-htpNat jSMwyr"
      onClick={[Function]}
      src={
        Object {
          "process": [Function],
        }
      }
    />
    <img
      alt="favorite"
      className="sc-htpNat jSMwyr"
      onClick={[Function]}
      src={
        Object {
          "process": [Function],
        }
      }
    />
  </div>
</div>
`;
